src/Stratis.Bitcoin.Features.SmartContracts/SmartContractPoABlockDefinition.cs

```diff
@@ -0,0 +1,50 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using Microsoft.Extensions.Logging;
+using NBitcoin;
+using Stratis.Bitcoin.Configuration;
+using Stratis.Bitcoin.Consensus;
+using Stratis.Bitcoin.Features.Consensus.CoinViews;
+using Stratis.Bitcoin.Features.MemoryPool;
+using Stratis.Bitcoin.Features.MemoryPool.Interfaces;
+using Stratis.Bitcoin.Features.Miner;
+using Stratis.Bitcoin.Features.PoA.ConsensusRules;
+using Stratis.Bitcoin.Utilities;
+using Stratis.SmartContracts.Core;
+using Stratis.SmartContracts.Core.State;
+using Stratis.SmartContracts.Core.Util;
+
+namespace Stratis.Bitcoin.Features.SmartContracts
+{
+    /// <summary>
+    /// Pushes everything to the <see cref="SmartContractBlockDefinition"/>, just amends the block difficulty for PoA.
+    /// </summary>
+    public class SmartContractPoABlockDefinition : SmartContractBlockDefinition
```

Comment:
Because there's more functionality in the SCBlockDefinition, it makes more sense to derive from it and add the PoA code rather than the other way around.

Unfortunately this sucks because sometimes we derive from the PoA class and sometimes we derive from the SC class